app/controllers/application_controller.rb

```diff
@@ -1,27 +1,33 @@
+# encoding: utf-8
+# ApplicationController inherits from ActionController::Base and
+# all other controllers inherits from it. It allows the configuration
+# of things such as request forgery protection and filtering of
+# sensitive request parameters.
+#
 class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
   helper_method :current_user, :logged_in?
-  
+
   def current_user
     @current_user ||= User.find_by uid: session[:user_id] if session[:user_id]
   end
-  
+
   def logged_in?
-    !!current_user 
+    !current_user.nil?
```

Comment:
@VivianChu this PR is excellent, great job. I was hoping to have a chance to review, but since it's already merged I'll just note that you could consider the Rails #present? method here instead of the double negative. Or, even more simply, I think just the following 'should' work but would need to be confirmed:

```
  def logged_in?
    current_user
  end
```

```
  def logged_in?
    current_user.present?
  end
```